tools/lint-md/lint-md.mjs

```diff
@@ -35,13 +35,6 @@ paths.forEach(async (path) => {
   const result = await linter.process(file);
   const isDifferent = fileContents !== result.toString();
 
-  if (path.startsWith('doc/api/')) {
-    if (!fileContents.includes('introduced_in')) {
-      console.error(`${path} is missing an 'introduced_in' version. Please add one.`);
-      process.exitCode = 1;
-    }
-  }
-
```

Comment:
Why this change?

Reply:
The api docs tooling has a linter that covers this check

Reply:
Markdown docs are still going to live in this repo, right? If so, I think we should keep the linters here

Reply:
The linter in the api docs tooling is being ran with `make lint-md`, or unless you're talking about keeping the code for the linting checks here?

Reply:
Why would we move the code for the checks?

Reply:
Because it makes sense to (at least imo),

The main thing is that the linting checks in the new tooling are easier to write and maintain since the parsing code is already taken care of. The checks only need to cover what they're looking for.

This one is more subjective, but, imo it also just makes sense to have tooling for the api docs with the other bits of tooling for the api docs

Reply:
The downside of removing it is that we don't know if it still holds. I'd be more comfortable to land with the checks, so the CI can confirm nothing has been broken, and in a follow-up we can remove each check with a reference to where the check were moved.

Reply:
sounds good to me